### ISSUE TITLE
Extending Buildings broke in latest build

### ISSUE THREAD

**Issue by sk7725** [Anuken/Mindustry#2548]
**Platform**: *Android/iOS/Mac/Windows/Linux*

**Build**: *9668*

**Issue**: *A working code, a library that implements EXP and leveling up, that automatically extends the block and building is somehow broken in the latest build - it works fine in 9652 and I cannot find out the problem at all. I suspect https://github.com/Anuken/Mindustry/commit/b58b6c7361ec1025500d755b97fbd8499752772f but I'm just lost.*

**Steps to reproduce**: *Code is here: [exp-lib](https://github.com/sk7725/EXP-lib/blob/master/scripts/library.js) but almost every library that extended the buildings have been broken. Crash report says the functions that were extended are non-existant.*

**Link(s) to mod(s) used**: *https://github.com/sk7725/EXP-lib*

**Save file**: *Irrelevant.*

**Crash report**: *From a mod that uses this library:
rhino.EcmaError: TypeError: Cannot find function totalExp in object Building#25. (exp#147)
	at rhino.ScriptRuntime.constructError(ScriptRuntime.java:3955)
	at rhino.ScriptRuntime.constructError(ScriptRuntime.java:3935)
	at rhino.ScriptRuntime.typeError(ScriptRuntime.java:3964)
	at rhino.ScriptRuntime.typeError2(ScriptRuntime.java:3980)
	at rhino.ScriptRuntime.notFunctionError(ScriptRuntime.java:4043)
	at rhino.ScriptRuntime.getPropFunctionAndThisHelper(ScriptRuntime.java:2350)
	at rhino.ScriptRuntime.getPropFunctionAndThis(ScriptRuntime.java:2333)
	at rhino.optimizer.OptRuntime.callProp0(OptRuntime.java:69)
	at rhino.gen.exp_13._c_anonymous_15(exp:147)
	at rhino.gen.exp_13.call(exp)
	at rhino.ArrowFunction.call(ArrowFunction.java:33)
	at rhino.Context.lambda$call$0(Context.java:515)
	at rhino.Context.call(Context.java:524)
	at rhino.Context.call(Context.java:515)
	at rhino.JavaAdapter.callMethod(JavaAdapter.java:549)
	at adapter47.get(<adapter>)
	at mindustry.ui.Bar.<init>(Bar.java:32)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at rhino.MemberBox.newInstance(MemberBox.java:137)
	at rhino.NativeJavaClass.constructInternal(NativeJavaClass.java:251)
	at rhino.NativeJavaClass.constructSpecific(NativeJavaClass.java:192)
	at rhino.NativeJavaClass.construct(NativeJavaClass.java:155)
	at rhino.ScriptRuntime.newObject(ScriptRuntime.java:2451)
	at rhino.gen.exp_13._c_anonymous_12(exp:143)
	at rhino.gen.exp_13.call(exp)
	at rhino.ContextFactory.doTopCall(ContextFactory.java:346)
	at rhino.ScriptRuntime.doTopCall(ScriptRuntime.java:3303)
	at rhino.gen.exp_13.call(exp)
	at rhino.ArrowFunction.call(ArrowFunction.java:33)
	at rhino.JavaAdapter.doCall(JavaAdapter.java:572)
	at rhino.JavaAdapter.callMethod(JavaAdapter.java:554)
	at adapter45.get(<adapter>)
	at mindustry.gen.Building.displayBars(Building.java:243)
	at mindustry.world.blocks.defense.turrets.ItemTurret$ItemTurretBuild.displayBars(ItemTurret.java:77)
	at mindustry.gen.Building.lambda$display$10(Building.java:1289)
	at arc.scene.ui.layout.Table.table(Table.java:244)
	at mindustry.gen.Building.display(Building.java:1287)
	at mindustry.ui.fragments.PlacementFragment.lambda$build$20(PlacementFragment.java:349)
	at arc.scene.ui.layout.Cell.lambda$update$0(Cell.java:154)
	at arc.scene.Element.act(Element.java:81)
	at arc.scene.Group.act(Group.java:37)
	at arc.scene.Group.act(Group.java:41)
	at arc.scene.Group.act(Group.java:41)
	at arc.scene.Group.act(Group.java:41)
	at arc.scene.Group.act(Group.java:41)
	at arc.scene.Group.act(Group.java:41)
	at arc.scene.Scene.act(Scene.java:170)
	at arc.scene.Scene.act(Scene.java:121)
	at mindustry.core.UI.update(UI.java:133)
	at arc.ApplicationCore.update(ApplicationCore.java:36)
	at mindustry.ClientLauncher.update(ClientLauncher.java:151)
	at arc.backend.sdl.SdlApplication.listen(SdlApplication.java:170)
	at arc.backend.sdl.SdlApplication.loop(SdlApplication.java:158)
	at arc.backend.sdl.SdlApplication.<init>(SdlApplication.java:52)
	at mindustry.desktop.DesktopLauncher.main(DesktopLauncher.java:36)*

---

*Place an X (no spaces) between the brackets to confirm that you have read the line below.*  
- [X] **I have searched the closed and open issues to make sure that this problem has not already been reported.**

**Comment by sk7725** [comment]
TL;DR: I extended Building to have `totalExp()` function, but at the latest build said extended Building's `totalExp()` is nowhere to be seen, despite being extended successfully. (`build.totalExp()` now crashes the game whereas it functioned as normal before)

**Comment by Anuken** [comment]
`entityType` and similar fields were renamed. The code isn't valid.
